site/content/docs/08-link-options.md

```diff
@@ -31,3 +31,15 @@ Adding a `rel=external` attribute to a link...
 ```
 
 ...will trigger a browser navigation when the link is clicked.
+
+### sapper-noscroll
+
+When navigating to internal links, Sapper will change the scroll position to 0,0 so that the user is at the very top left of the page. When a hash is defined, it will scroll to the element with a matching ID.
+
+In certain cases, you may wish to disable this behaviour. Adding a `sapper-noscroll` attribute to a link...
+
+```html
+<a href='path' sapper-noscroll>Path</a>
```

Comment:
Personally I'd prefer `noscroll` attribute. Maybe others can weigh in?
```suggestion
<a href='path' noscroll>Path</a>
```

Reply:
@lukeed agree

Reply:
`stickywhenclicky` imo

Reply:
😂

Reply:
This change is not relevant to this PR — I only documented why currently exists

Reply:
Prefixing with sapper is obviously the safer option, and likely was done to avoid any risk of colliding with future spec changes. I think this discussion should be taken to an issue

Reply:
Correct, I'm asking for input before we officially document something

Reply:
Fair enough.

Personally I am wary of adding arbitrary attributes to DOM elements without some kind of namespace, as it could lead to collisions in the future and with little benefit (except perhaps looking a little cleaner). For this reason I’d have actually used a data attribute (which also would make this valid markup)

Reply:
Some options:
* `sapper-noscroll` (verbose. but would avoid breaking change)
* `noscroll` (nice looking, but not namespaced)
*  `s:noscroll` (angular-like. perhaps svelte would want s namespace in the future though?)
*  `s-noscroll` (vue-like. perhaps svelte would want s namespace in the future though?)
*  `data-noscroll` (valid markup)
*  `data-s-noscroll` (valid markup and namespaced, but verbose)
*  `data-s:noscroll` (valid markup and namespaced, but verbose)

Probably `s-noscroll` and `data-s:noscroll` would be my favorites. I like that the latter is valid markup. It's a bit more verbose though. Right now that's probably fine because I don't know how much this option will be used, but maybe that'd be more annoying if we add more frequently used options with the same syntax?

Reply:
What about `sapper:noscroll`?
https://github.com/sveltejs/sapper/blob/ca3645768597035b08e9d0e025cabd062180184a/runtime/src/app/start/index.ts#L113
Since it's not accessed through `.dataset`, I think having a `data-` prefix (even though valid HTML) is redundant

Reply:
The fact that it *could* be accessed via dataset & also via a data attribute selector in css (and is valid HTML) are all strong points for. Are there any points against other than aesthetics/verbosity?

Reply:
As a side note to the actual decision — to change this will be a breaking change, despite it being previously undocumented. I am not sure that it is reasonable to wrap a breaking change up with my PR here. Perhaps I should remove the documentation of `sapper-noscroll` from this PR, and make a new one where this discussion can continue?

Reply:
I think I prefer `sapper-noscroll` which makes it clear that it's a Sapper specific thing. 'noscroll' feels like an built-in attribute.

Reply:
I took a poll of the maintainers offline and there was a preference for keeping noscroll namespaced. However, folks preferred `sapper:noscroll` to `sapper-noscroll`, so I updated this PR to rename the attribute.

I'll go ahead and merge this now. Thanks for the patience and input everyone!

Reply:
I also prefer `sapper:noscroll`, but I'm curious, are there other namespaced attributes for html elements?

Reply:
Not currently, but that's where things are going to be moving in the future. `rel="prefetch"` (which, when used on an anchor element, is a Sapper-only thing, which is confusingly named the same as a real HTML thing) is going to be renamed to some other namespaced `sapper:` attribute.

Reply:
Great, I like this idea. Much clearer!